utils/src/main/java/org/corfudb/utils/lock/persistence/LockStore.java

```diff
@@ -90,7 +93,11 @@ public LockStore(CorfuRuntime runtime, UUID clientUuid) throws NoSuchMethodExcep
      * @throws LockStoreException
      */
     public boolean acquire(LockId lockId) throws LockStoreException {
-        CorfuStoreMetadata.Timestamp timestamp = corfuStore.getTimestamp();
+        Token token = runtime.getSequencerView().query().getToken();
+        CorfuStoreMetadata.Timestamp timestamp = CorfuStoreMetadata.Timestamp.newBuilder()
+                .setEpoch(token.getEpoch())
+                .setSequence(token.getSequence())
+                .build();
```

Comment:
LockStore still use the deprecated query API, so TS is still here. Do we plan to remove it?

Just wanted to make sure my understand is correct - Every corfu store user should get timestamp from read txn commit, rather than querying token to build this TS. And looks like user should not have its runtime?

Reply:
correct, actually I'm not sure why this TS was needed to be passed on the update and create if it does not need a snapshot. I completely removed the dependency from timestamp, please take a look in case I missed something.